src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -135,6 +149,21 @@ public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer)
             indexCreator = () -> createKNNIndexFromScratch(field, pair, knnEngine, indexPath);
         }
 
+        if (isMerge) {
+            KNNGraphValue.MERGE_CURRENT_OPERATIONS.set(KNNGraphValue.MERGE_CURRENT_OPERATIONS.getValue() - 1);
+            KNNGraphValue.MERGE_CURRENT_DOCS.set(KNNGraphValue.MERGE_CURRENT_DOCS.getValue() - pair.docs.length);
+            KNNGraphValue.MERGE_CURRENT_SIZE_IN_BYTES.set(
+                KNNGraphValue.MERGE_CURRENT_SIZE_IN_BYTES.getValue() - calculateArraySize(pair.vectors)
+            );
+            KNNGraphValue.MERGE_TOTAL_OPERATIONS.increment();
+            KNNGraphValue.MERGE_TOTAL_DOCS.set(KNNGraphValue.MERGE_TOTAL_DOCS.getValue() + pair.docs.length);
+            KNNGraphValue.MERGE_TOTAL_SIZE_IN_BYTES.set(KNNGraphValue.MERGE_TOTAL_SIZE_IN_BYTES.getValue() + arraySize);
+        }
+
+        if (isRefresh) {
+            KNNGraphValue.REFRESH_TOTAL_OPERATIONS.increment();
+        }
```

Comment:
can we abstract this code as 2 separate function, which says recordMergeStats and recordRefreshStats?

Also, in refresh don't we get information around how many documents got refreshed similar to merge?

Reply:
We do get that information, but that information is not present on the OpenSearch node stats so I didn't include it